resim.py

```diff
@@ -895,289 +898,102 @@ def handle_out_advances(self, fielder):
-                ((2, 1, 0), (2, 1, 0)): 2,  # !DP roll (fail), !martyr roll (fail)
-                ((2, 1, 2, 0), (2, 0)): 3,  # ?DP roll (fail), ?martyr roll(fail), ???
-            }
+            if len(self.update["basesOccupied"]) > 0:
+                dp_roll = self.roll("dp?")
 
-            event_type = "Out"
-            if "reaches on fielder's choice" in self.desc:
-                # !DP roll (fail), !martyr roll (fail)
-                extras[((2, 0), (0,))] = 2  # what
-                event_type = "FC"
-
-            if "into a double play!" in self.desc:
-                # !DP roll (pass), !roll<0.33 out at home
-                extras[((2, 1, 0), (2, 1))] = 2
-                event_type = "DP"
-
-            extra_roll_desc = extras[
-                (
-                    tuple(self.update["basesOccupied"]),
-                    tuple(self.next_update["basesOccupied"]),
-                )
-            ]
-            extra_rolls = [self.roll("extra") for _ in range(extra_roll_desc)]
-
-            # DP rolls
-            if 0 in self.update["basesOccupied"]:
-                self.log_roll(Csv.GROUNDOUT_FORMULAS, "DP", extra_rolls[0], event_type == "DP", fielder=fielder)
-
-            # Martyr/Sacrifice rolls
-            if (
-                ((self.update["basesOccupied"] in [[0]]) and (self.next_update["basesOccupied"] in [[0], [1]]))
-                or (
-                    (self.update["basesOccupied"] in [[1, 0]])
-                    and (self.next_update["basesOccupied"] in [[1, 0], [2, 1]])
-                )
-                or (
-                    (self.update["basesOccupied"] in [[2, 0]])
-                    and (self.next_update["basesOccupied"] in [[0], [1], [2, 1]])
-                )
-                or (
-                    (self.update["basesOccupied"] in [[2, 1, 0]])
-                    and (self.next_update["basesOccupied"] in [[2, 1], [2, 1, 0]])
-                    and (event_type != "DP")
-                )
-            ):
+                if 0 in self.update["basesOccupied"]:
+                    is_dp = "into a double play!" in self.desc
+                    self.log_roll(Csv.GROUNDOUT_FORMULAS, "DP", dp_roll, is_dp, fielder=fielder)
 
-                passed = (
-                    ((self.update["basesOccupied"] in [[0]]) and (self.next_update["basesOccupied"] in [[1]]))
-                    or ((self.update["basesOccupied"] in [[1, 0]]) and (self.next_update["basesOccupied"] in [[2, 1]]))
-                    or ((self.update["basesOccupied"] in [[2, 0]]) and (len(extra_rolls) == 4))
-                    or (
-                        (self.update["basesOccupied"] in [[2, 1, 0]])
-                        and (self.next_update["basesOccupied"] in [[2, 1]])
-                    )
-                )
+                    if is_dp:
+                        self.roll("dp where")  # (index into basesOccupied)
```

Comment:
Should we add a check to see if this the roll matches the expected outcome? I assume not doing so is more in line with how you do everything else.